### PR TITLE
detect and ignore doubled trigger edges

### PR DESCRIPTION
**Pull request by mck1117**
Detect when we get a "doubled edge" trigger input.

Will enable the second check (very wrong tooth timing) later, as it causes problems at the moment.

#987 #2222 #4650

### CODE REVIEW

**Comment by mck1117** [comment]
this looks also OK, proteus CI still sad 🙁

**Comment by rusefillc** [comment]
What do we perceive to be root cause here? Is it expected to get double edges or it is some specific hardware issue?

Do we want to changelog this improvement?

File: firmware/controllers/trigger/trigger_central.cpp [hunk 1]
```diff
@@ -641,6 +641,62 @@ void TriggerCentral::decodeMapCam(efitick_t timestamp, float currentPhase) {
 	}
 }
 
+bool TriggerCentral::isToothExpectedNow(efitick_t timestamp) {
```
Comment (rusefillc) [inline firmware/controllers/trigger/trigger_central.cpp:644]: See open comments on ticket

Please rebase to get green build

**Review by rusefillc — CHANGES_REQUESTED** [review]

**Comment by rusefillc** [comment]
@ElDominio tell me about 4b11 is it Hall or VR?

**Comment by rusefillc** [comment]
@mck1117 please blink if alive

**Comment by mck1117** [comment]
I'm alive! Was racing this weekend and getting ready before that.

**Comment by rusefillc** [comment]
What do we perceive to be root cause here? Is it expected to get double edges or it is some specific hardware issue?

Do we want to changelog this improvement?

**Comment by rusefillc** [comment]
@mck1117 is this abandoned?

**Comment by ElDominio** [comment]
My only question with this is how do some other ECU companies trigger properly on both edges (no crazy sync loss)

**Comment by mck1117** [comment]
> My only question with this is how do some other ECU companies trigger properly on both edges (no crazy sync loss)

sort of unrelated discussed offline re: #4635

**Comment by mck1117** [comment]
> What do we perceive to be root cause here? Is it expected to get double edges or it is some specific hardware issue?

It is a case we can detect cheaply, so why not detect it? Perhaps it will be a useful metric to detect issues we know about but haven't narrowed down to a root cause.

**Comment by rusefillc** [comment]
> > What do we perceive to be root cause here? Is it expected to get double edges or it is some specific hardware issue?
> 
> It is a case we can detect cheaply, so why not detect it? Perhaps it will be a useful metric to detect issues we know about but haven't narrowed down to a root cause.

I like this justification! My preference would be to have this filtering enabled by default but have an option to turn it off, what do you think? I am fine adding the option later myself.

**Comment by mck1117** [comment]
> I like this justification! My preference would be to have this filtering enabled by default but have an option to turn it off, what do you think? I am fine adding the option later myself.

Right now it only filters truly doubled edges (no trigger should have 1/2 degree tooth spacing), but sure

**Pull request merged**